addons/html_builder/static/src/plugins/version_control_option.js

```diff
@@ -0,0 +1,30 @@
+import { BaseOptionComponent } from "@html_builder/core/utils";
+
+export class VersionControlOption extends BaseOptionComponent {
+    static template = "html_builder.VersionControlOption";
+    static props = {
+        accessPerOutdatedEl: Object,
+    };
+    setup() {
+        super.setup();
+        this.snippets = this.env.services["html_builder.snippets"];
+        const el = this.env.getEditingElement();
+        const snippetKey = el.dataset.snippet;
+        const snippet = this.snippets.getOriginalSnippet(snippetKey);
+        const {
+            vcss: originalVcss,
+            vxml: originalVxml,
+            vjs: originalVjs,
+        } = snippet.content.dataset;
+        const { vcss: elVcss, vxml: elVxml, vjs: elVjs } = el.dataset;
+        const isAccessingOutdated = this.props.accessPerOutdatedEl.get(el);
+        this.env.optionContainerState.isUpToDate =
+            isAccessingOutdated ||
+            (originalVcss === elVcss && originalVxml === elVxml && originalVjs === elVjs);
```

Comment:
It s not a good practice to edit the state of another comp. What do you want to do ?